typescript/src/ethereum.ts

```diff
@@ -367,7 +393,11 @@ export class Bridge
       vault: vault ? `0x${vault.identifierHex}` : constants.AddressZero,
     }
 
-    const tx = await this._instance.revealDeposit(depositTxParam, revealParam)
+    const tx = await backoffRetrier<ContractTransaction>(
+      this._totalRetryAttempts
+    )(async () => {
+      return await this._instance.revealDeposit(depositTxParam, revealParam)
+    })
```

Comment:
Will we retry in case it is not an unexpected failure (e.g. broken connection) but the gas estimation fails because the deposit has been already revealed? Ideally, I think we shouldn't retry for the latter but it's hard to implement and the current solution is an improvement anyway so I do not find it blocking.

Reply:
I pushed a couple of commits that should improve it and cover the situation you mentioned:
ccd4d0a9b2a6be5392151cd5c7dee259ef89f69c, 5e19011be61bf8b2009e3d67ad4765bc37440a8e, 0cda5e7a587432774e53624a839a4abb153531f6